src/thing_stats.c

```diff
@@ -403,6 +403,36 @@ long compute_creature_max_armour(long base_param, unsigned short crlevel, TbBool
   return max_param;
 }
 
+/**
+ * Computes training cost of a creature on given level.
+ */
+long compute_creature_max_training_cost(long base_param,unsigned short crlevel)
```

Comment:
where possible it's preferred to use appropriate typedefs, in this case GoldAmount

Reply:
I copied the function for payday, should I use GoldAmount for all three functions that compute cost?

Is it like this?
GoldAmount compute_creature_max_training_cost(long base_param,unsigned short crlevel)

or like this?
long compute_creature_max_training_cost(GoldAmount base_param,unsigned short crlevel)

Reply:
> where possible it's preferred to use appropriate typedefs, in this case GoldAmount

I copied the function for payday, should I use GoldAmount for all three functions that compute cost?

Is it like this?
GoldAmount compute_creature_max_training_cost(long base_param,unsigned short crlevel)

or like this?
long compute_creature_max_training_cost(GoldAmount base_param,unsigned short crlevel)

Reply:
> where possible it's preferred to use appropriate typedefs, in this case GoldAmount

Ok so I did that on multiple instance, please tell me if it was unnecessary on some place or is it fine now?

Reply:
> where possible it's preferred to use appropriate typedefs, in this case GoldAmount

Ok so I did replace long with GoldAmount on multiples places, please tell me if it was unnecessary to do so or is it fine now? 🤔